pkg/allocator/allocator.go

```diff
@@ -202,11 +202,15 @@ type Backend interface {
 	// error in that case is not expected to be fatal. The actual ID is obtained
 	// by Allocator from the local idPool, which is updated with used-IDs as the
 	// Backend makes calls to the handler in ListAndWatch.
-	AllocateID(ctx context.Context, id idpool.ID, key AllocatorKey) error
+	// The implementation of the backend might return an AllocatorKey that is
+	// a copy of 'key' with an internal reference of the backend key.
```

Comment:
I'd suggest to slightly rephrase this comment, as I personally read it as "sometimes the AllocatorKey might be nil". Instead, IIUC, it is only the `BackendKey` which might not be set (unless a non-nil error is returned, which is expected).